lib/src/fileset.rs

```diff
@@ -217,6 +226,18 @@ impl FilesetExpression {
         }
     }
 
+    /// Iterates literal paths in the top-level expression. This doesn't search
```

Comment:
So IIUC, because this doesn't recurse into Intersection or Difference, a future expression like "lib/src & glob(**.rss)" won't warn that it didn't match anything under "lib/src".  Would it be reasonable to keep a 'negated' bool, and recurse into Intersection and Difference expressions appropriately?

Reply:
Correct. It's just a typo checking, so I think it's okay to ignore compound expressions.

Alternatively, file paths could be "resolved" in a similar way to revset symbols, but doing that wouldn't be simple.

Reply:
I think you can still check compound expressions pretty easily here without doing any advanced resolution.  You could rewrite this method as:

pub fn visit_nonnegative_paths(name?)(&self, visitor: impl Fn(&RepoPath)) {
  match self {
    ...
    Intersection(a, b) => { a.visit_nonnegative_paths(...); b.visit_nonnegative_paths(...); }
    Difference(a, b) => { a.visit_nonnegative_paths();  // Skip b }
  }
}

Reply:
Yeah, callback or something collecting into `Default + Extend` can be easily implemented.

Regarding negative expressions, it might be actually better to check existence of negated paths. If negative path is used in `jj split`, the user would expect the pattern  matches at least one file.

Reply:
Reimplemented as a recursive iterator. PTAL.

Reply:
Thanks!